cclient/serialization/json/get_balances.c

```diff
@@ -33,22 +33,94 @@ retcode_t json_get_balances_serialize_request(
 
   cJSON_AddItemToObject(json_root, "threshold",
                         cJSON_CreateNumber(obj->threshold));
+  if (obj->tips) {
+    ret = hash243_queue_to_json_array(obj->tips, json_root, "tips");
+    if (ret != RC_OK) {
+      goto err;
+    }
+  }
+
+  json_text = cJSON_PrintUnformatted(json_root);
+  if (json_text) {
+    len = strlen(json_text);
+    ret = char_buffer_allocate(out, len);
+    if (ret == RC_OK) {
+      strncpy(out->data, json_text, len);
+    }
+  }
+
+  return ret;
 
-  ret = hash243_queue_to_json_array(obj->tips, json_root, "tips");
+err:
+  cJSON_Delete(json_root);
+  return ret;
+}
+
+retcode_t json_get_balances_deserialize_request(serializer_t const *const s,
+                                                char const *const obj,
+                                                get_balances_req_t *const req) {
+  retcode_t ret = RC_OK;
+  cJSON *json_obj = cJSON_Parse(obj);
+  cJSON *json_item = NULL;
+  log_info(json_logger_id, "[%s:%d] %s\n", __func__, __LINE__, obj);
+
+  JSON_CHECK_ERROR(json_obj, json_item, json_logger_id);
+
+  ret = json_get_uint8(json_obj, "threshold", &req->threshold);
+  if (ret) {
+    goto end;
+  }
+
+  ret = json_array_to_hash243_queue(json_obj, "addresses", &req->addresses);
+  if (ret) {
+    goto end;
+  }
+  if (cJSON_GetObjectItemCaseSensitive(json_obj, "tips")) {
+    ret = json_array_to_hash243_queue(json_obj, "tips", &req->tips);
```

Comment:
`json_array_to_hash243_queue` already checks for whether the item exists & will return an error code if it doesn't, so no need to wrap this in an `if`